### DISCUSSION TITLE
Question about shouldLogEvent() method

### DISCUSSION THREAD

**Discussion by reppair** [spatie/laravel-activitylog#986] [Q&A]
Hi there, a question here about the `shouldLogEvent()` method definition.

Here is how the function looks in `v4`..

```php
    protected function shouldLogEvent(string $eventName): bool
    {
        $logStatus = app(ActivityLogStatus::class);

        if (! $this->enableLoggingModelsEvents || $logStatus->disabled()) {
            return false;
        }

        if (! in_array($eventName, ['created', 'updated'])) {
            return true;
        }

        $deletedAtColumn = method_exists($this, 'getDeletedAtColumn')
            ? $this->getDeletedAtColumn()
            : 'deleted_at';

        if (Arr::has($this->getDirty(), $deletedAtColumn)) {
            if ($this->getDirty()[$deletedAtColumn] === null) {
                return false;
            }
        }

        // Do not log update event if only ignored attributes are changed.
        return (bool) count(Arr::except($this->getDirty(), $this->activitylogOptions->dontLogIfAttributesChangedOnly));
    }
```

What is the purpose of `if ($this->getDirty()[$deletedAtColumn] === null) return false`?

I have a factory that defines `'deleted_at' => null` as default model state, in turn, when I seed my model's table there are no `created` logs for it due to that check.

Maybe we can improve on that, that's why I am asking. Thanks.

**Comment by Gummibeer** [comment]
Setting the deleted column directly isn't the recommended way to restore a model. But use the `restore()` method which will trigger the `restore` event which is logged separated.

**Reply by reppair** [reply]
Missing the point, I have a factory which declares it null as default for new models..
